src/http.rs

```diff
@@ -556,10 +995,19 @@ where
 
 #[cfg(test)]
 mod tests {
```

Comment:
Nit: Let's extract these into their own `tests.rs` because this file is getting too large.

Reply:
what to do with method like 
```rust
fn default_request_headers<B>(request: &mut Request<B>, user_agent: &HeaderValue) {
    if !request.headers().contains_key("User-Agent") {
        request
            .headers_mut()
            .insert("User-Agent", user_agent.clone());
    }

    if !request.headers().contains_key("Accept-Encoding") {
        // hardcoding until we support compressed responses:
        // https://github.com/vectordotdev/vector/issues/5440
        request
            .headers_mut()
            .insert("Accept-Encoding", HeaderValue::from_static("identity"));
    }
}
```
make it public or copy ?

Reply:
We could make them public. But since this PR is getting too big, feel free to ignore this comment. Resolving this thread.